packages/liveblocks-core/src/room.ts

```diff
@@ -882,16 +890,29 @@ export function createRoom<
   const batchUpdates = config.unstable_batchedUpdates ?? doNotBatchUpdates;
 
   function onStatusDidChange(newStatus: PublicConnectionStatus) {
+    const token = managedSocket.token?.parsed;
+    const sessionInfo = token
+      ? {
+          id: token.actor,
+          userInfo: token.info,
+          userId: token.id,
+          isReadOnly: isStorageReadOnly(token.scopes),
+        }
+      : null;
+
     if (newStatus === "open" || newStatus === "connecting") {
+      if (sessionInfo === null) {
+        throw new Error("Unexpected missing session info");
+      }
       context.connection.set({
         status: newStatus,
-        id: managedSocket.token.parsed.actor,
-        userInfo: managedSocket.token.parsed.info,
-        userId: managedSocket.token.parsed.id,
-        isReadOnly: isStorageReadOnly(managedSocket.token.parsed.scopes),
+        sessionInfo,
       });
     } else {
-      context.connection.set({ status: newStatus });
+      context.connection.set({
+        status: newStatus,
+        lastSessionInfo: sessionInfo,
```

Comment:
This was the crucial missing piece that caused the bug. When the connection dropped, the status was set to `status: "unavailable"`[^1] and the `sessionInfo` from the previously-`"open"` connection would be lost. Here, we're retaining it, but clearly marking it as "last" session info.

[^1]: Soon: `"reconnecting"`.